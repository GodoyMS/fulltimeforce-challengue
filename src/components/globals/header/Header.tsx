import { Image, TouchableOpacity, View } from "react-native";
import React from "react";
import { dispatch } from "@navigations/rootNavigations";
import { DrawerActions } from "@react-navigation/native";
import { AntDesign, MaterialCommunityIcons } from "@expo/vector-icons";
import { COLORS, SIZES } from "@constants/theme";
import MenuImg from "@assets/images/icons/menu.png"
import { useDrawerStatus } from "@react-navigation/drawer";
import { useThemeContext } from "@context/themeContext";
const Header = () => {
  const statusDrawer=useDrawerStatus()
  const{setColors}=useThemeContext()
  if(statusDrawer==="open")return
  return (
    <View
      style={{
        justifyContent: "space-between",
        alignItems: "center",
        flexDirection: "row",
        width: "100%",
        paddingHorizontal: SIZES.horizontal,
      }}
    >
      <TouchableOpacity
        onPress={() => {
          dispatch(DrawerActions.toggleDrawer());
          setColors([COLORS.primary.low,COLORS.primary.low])
        }}
        style={{}}
      >
        <Image
        style={{width:24,height:24}}
          source={MenuImg}
        />
      </TouchableOpacity>

      <TouchableOpacity
        
        style={{}}
      >
        <MaterialCommunityIcons name="filter-outline" size={24} color="white" />
      </TouchableOpacity>
    </View>
  );
};

export default Header;
