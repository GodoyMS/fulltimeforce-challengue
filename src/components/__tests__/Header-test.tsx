import React from "react";
import renderer from "react-test-renderer";
import { DrawerActions } from "@react-navigation/native";
import Header from "@components/globals/header/Header";
import { useDrawerStatus } from "@react-navigation/drawer";

// Mock theme context
jest.mock("@context/themeContext", () => ({
  useThemeContext: () => ({
    setColors: jest.fn(),
  }),
}));

// Mock drawer status
jest.mock("@react-navigation/drawer", () => ({
  useDrawerStatus: jest.fn(() => "closed"),
}));

// Mock root navigation dispatch
jest.mock("@navigations/rootNavigations", () => ({
  dispatch: jest.fn(),
}));

// Mock icon from expo
jest.mock("@expo/vector-icons", () => ({
  AntDesign: () => "AntDesign",
  MaterialCommunityIcons: () => "MaterialCommunityIcons",
}));


describe("Header component", () => {
    it("renders correctly when drawer is closed", () => {
        // Mock drawer status to be closed
        const useDrawerStatus = require("@react-navigation/drawer").useDrawerStatus;
        useDrawerStatus.mockReturnValueOnce("close");
          
        const tree = renderer.create(<Header />).toJSON();
        expect(tree).toBeDefined();
      });
    
  it("renders nothing when drawer is open", () => {
    // Change mock drawer status to "open"
    const useDrawerStatus = require("@react-navigation/drawer").useDrawerStatus;
    useDrawerStatus.mockReturnValueOnce("open");
  
    const tree = renderer.create(<Header />).toJSON();
    expect(tree).toBeNull(); // Because the component returns null when drawer is open
  });
  

});
