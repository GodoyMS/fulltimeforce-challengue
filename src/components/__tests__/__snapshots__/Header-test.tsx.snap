// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component renders correctly when drawer is closed 1`] = `
<View
  accessibilityIgnoresInvertColors={true}
  style={
    [
      {
        "paddingBottom": 40,
        "paddingHorizontal": 32,
        "paddingTop": 96,
      },
      {
        "backgroundColor": "#F3F3F3",
      },
    ]
  }
>
  <Image
    accessibilityRole="image"
    source={
      {
        "testUri": "../Libraries/NewAppScreen/components/logo.png",
      }
    }
    style={
      [
        {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        {
          "height": undefined,
          "width": undefined,
        },
        {
          "marginBottom": -192,
          "marginLeft": -128,
          "opacity": 0.2,
          "overflow": "visible",
          "resizeMode": "cover",
        },
      ]
    }
    testID="new-app-screen-header"
  />
  <Text
    style={
      [
        {
          "fontSize": 40,
          "fontWeight": "700",
          "textAlign": "center",
        },
        {
          "color": "#000",
        },
      ]
    }
  >
    Welcome to
    

    React Native
  </Text>
</View>
`;

exports[`Header component renders correctly when drawer is closed 2`] = `
<View
  style={
    {
      "alignItems": "center",
      "flexDirection": "row",
      "justifyContent": "space-between",
      "paddingHorizontal": 30,
      "width": "100%",
    }
  }
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
  >
    <Image
      source={
        {
          "testUri": "../../../src/assets/images/icons/menu.png",
        }
      }
      style={
        {
          "height": 24,
          "width": 24,
        }
      }
    />
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
  >
    MaterialCommunityIcons
  </View>
</View>
`;

exports[`renders ThemedBadge correctly with a title 1`] = `
<View
  style={{}}
>
  <ViewManagerAdapter_ExpoLinearGradient
    colors={
      [
        4279383126,
        4284826401,
      ]
    }
    style={
      {
        "borderRadius": 50,
        "padding": 10,
      }
    }
  >
    <Text
      style={
        [
          undefined,
          undefined,
          undefined,
          {
            "fontSize": 16,
          },
          undefined,
          undefined,
          undefined,
          undefined,
          {
            "fontFamily": "DMMedium",
          },
          undefined,
          {
            "color": "#ffffff",
          },
          undefined,
        ]
      }
    >
      Hello World
    </Text>
  </ViewManagerAdapter_ExpoLinearGradient>
</View>
`;
